gym_ignition/base/robot/robot_baseframe.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import numpy as np
+from typing import Tuple
+from abc import ABC, abstractmethod
+
+# TODO: new exception for not existing joints? Add Exceptions in the docs?
+
+
+class RobotBaseFrame(ABC):
+    """
+    Robot base frame interface.
+
+    This interface provides methods to get and set quantities related to the robot base.
+    """
+
+    def __init__(self) -> None: ...
+
+    @abstractmethod
+    def set_base_frame(self, frame_name: str) -> bool:
+        """
+        Set the base frame of the robot. If this method is not called, the detection of
+        the base frame depends on the implementation.
+
+        Args:
+            frame_name: The name of the base frame.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def base_frame(self) -> str:
+        """
+        Get the name of the base frame.
+
+        Returns:
+            The name of the base frame.
+        """
+
+    @abstractmethod
+    def base_pose(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Returns the pose of the robot base.
+
+        The pose is composed of a position and a quaternion associated to the base frame,
+        both expressed in the the world inertial frame.
+
+        Returns:
+            A Tuple containing the position and orientation of the base:
+
+            - position: 3D array in the [x, y, z] form.
+            - orientation: a 4D array containing a quaternion in the [w, x, y, z] form.
```

Comment:
To define in an unambiguous way the quaternion, it could me a nice idea to document how the quaternion is mapped to a {}^A R_B rotation matrix. This may go in a separate doc page, together with the definition of the link velocity.

Reply:
Personal opinion: for having a clear reference, an `.md` document in the repo is enough, and when you will have the website you can always take it an make more pretty.

Reply:
As above, I linked also this comment to https://github.com/robotology/gym-ignition/issues/37 as a reminder. Thanks!